File: examples/read.cxx
#include <fstream>
#include <iostream>
#include "nifty_common.pb.h"

int main() {
    nifty_common::v1::BoardState batch;

    std::string in_filename = "nifty_common.bin";
    std::ifstream in(in_filename, std::ios::binary);
    batch.ParseFromIstream(&in);

    std::cout << "Reading from file: " << in_filename << "\n";
    std::cout << "Version: " << batch.version() << "\n";
    std::cout << "Creator ID: " << batch.creator_id() << "\n";
    std::cout << "Number of nodes: " << batch.nodes().shape(0) << "\n";
    std::cout << "Node feature dimension: " << batch.nodes().shape(1) << "\n";
    std::cout << "Number of edges: " << batch.edge_index().shape(1) << "\n";


    return 0;
}

File: examples/read.py
import nifty_common_pb2 as pb
from utils import NiftyConverter

with open("nifty_common.bin", "rb") as f:
    batch = pb.BoardState()
    batch.ParseFromString(f.read())
    print("Nodes:", batch.nodes.shape, batch.nodes.dtype)
    print("Edge Index:", batch.edge_index.shape, batch.edge_index.dtype)
    print("Edge Attributes:", batch.edge_attributes.shape, batch.edge_attributes.dtype)

    converter = NiftyConverter()
    data = converter.to_pyg(batch)
    print("BoardState converted to PyTorch Geometric Data object:", data)
